src/snowflake/cli/app/dev/docs/templates/definition_description.rst.jinja2

```diff
@@ -0,0 +1,28 @@
+{{ title }}
+
+Structure
+===============================================================================
+
+.. code-block::
+
+{% for field in fields %}
+{{ "  "*(field["indents"] + 1) + ("- " if field["item_index"] == 1 else "") + ("  " if field["item_index"] > 1 else "") + field["name"] }}: {% if field["add_types"] %}<{{ field["types"] }}>{% endif %}
```

Comment:
Can we add also information if the option is optional? We can use comments, for example
```rst
  definition_version: <string | integer>
  snowpark: 
    project_name: <string>
    stage_name: <string>
    src: <string>
    functions: 
      - name: <string>
        handler: <string>
        returns: <string>
        signature: 
        - name: <string>
          type: <string>
          default: <string>  # optional
        database: <string>  # optional
        schema: <string>  # optional
```

WDYT?

Reply:
I think we should avoid adding the condition to the field, as it leads to information overload and complicates the understanding of the structure. I was thinking it would be clearer to use examples instead of each field type. 

This is how it would look like with nativeapp
 ```
definition_version: <string | integer>
  native_app: 
    name: <string>
    artifacts: 
      - src: <string>  # optional
        dest: <string>  # optional
        processors: 
        - name: <string>
          properties: <object>  # optional
    deploy_root: <string>  # optional
    generated_root: <string>  # optional
    source_stage: <string>  # optional
    scratch_stage: <string>  # optional
    package: 
      scripts: <array[string]>  # optional
      role: <string>  # optional
      name: <string>  # optional
      warehouse: <string>  # optional
      distribution: <string>  # optional
    application: 
      role: <string>  # optional
      name: <string>  # optional
      warehouse: <string>  # optional
      debug: <boolean>  # optional
```

Reply:
That a good comment, I agree